thavalon-server/src/connections/validation.rs

```diff
@@ -270,3 +270,159 @@ pub async fn validate_password(plaintext: &String, hash: &String) -> bool {
 
     result
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use chrono::{Duration, Utc};
+    use scrypt::ScryptParams;
+
+    /// Tests hashing passwords against an insecure password.
+    #[tokio::test]
+    async fn test_hash_password_insecure() {
+        let result = hash_password(&String::from("not_sec")).await;
+        assert!(result.unwrap_err() == ValidationError::InvalidPassword);
+    }
+
+    /// Tests hasing passwords against a secure password.
+    #[tokio::test]
+    async fn test_hash_password_normal_pw() {
+        let password = String::from("32fdsfjidsfj293423");
+        let result = hash_password(&password)
+            .await
+            .expect("Failed to hash password correctly.");
+        scrypt::scrypt_check(&password, &result).expect("Failed to match password hashes.");
+    }
+
+    /// Tests validating a password with a matching hash.
+    #[tokio::test]
+    async fn test_validate_password_match() {
+        let password = String::from("asdfwe322ef2342");
+        let hash = scrypt::scrypt_simple(&password, &ScryptParams::recommended()).unwrap();
+        assert!(validate_password(&password, &hash).await);
+    }
+
+    /// Tests validating a password with an invalid hash. This shouldn't match.
+    #[tokio::test]
+    async fn test_validate_password_bad_hash() {
+        let password = String::from("23qsadf2323f");
+        assert!(!validate_password(&password, &password).await);
+    }
+
+    /// Tests validating a password with a mismatched hash.
+    #[tokio::test]
+    async fn test_validate_password_mismatch() {
+        let password = String::from("32f23f2ef23");
+        let other_password = String::from("342f98j98j34gf");
+        let hash = scrypt::scrypt_simple(&other_password, &ScryptParams::recommended()).unwrap();
+        assert!(!validate_password(&password, &hash).await);
+    }
+
+    /// Tests creating a JWT for a given player ID. Expected results generated from jwt.io.
+    #[tokio::test]
+    async fn test_create_jwt_valid() {
+        let mut mananger = TokenManager::new();
+        let (jwt, _) = mananger.create_jwt(&String::from("TESTING_THIS")).await;
+        let expires_at = Utc::now()
+            .checked_add_signed(Duration::minutes(15))
+            .unwrap()
+            .timestamp();
+        assert!(expires_at - 60 <= jwt.expires_at);
+        assert!(jwt.expires_at <= expires_at + 60);
+        assert_eq!(jwt.token_type, "Bearer".to_string());
```

Comment:
nit: `jwt.token_type.as_str()` instead of creating a new String

Reply:
Fixed